### ISSUE TITLE
Issue on Synology

### ISSUE THREAD

**Issue by ukdtom** [ukdtom/WTInstaller#23]
https://github.com/ukdtom/WebTools.bundle/issues/530

Any idea @ChuckPa or @trumpy81 ?

**Comment by cswanghan** [comment]
any solutions? got the same problem here.

**Comment by ukdtom** [comment]
Sadly I don't have a Synology box, so we can only hope that either @ChuckPa or @trumpy81 can shear some light on this

**Comment by trumpy81** [comment]
The only thing that comes to mind is that the download of the actual package did not complete correctly.

Try uninstalling the package, manually if need be, then try to install it again.

Do not use Safari or Chrome when you attempt to install WebTools as they have been known to not complete downloads correctly. Although they may not be involved in this case.

Otherwise, do a full manual install by downloading the WebTools package from Github and then unpack it in the plugins folder.

https://github.com/ukdtom/WebTools.bundle click on clone or download and then select Download Zip.

**Comment by ChuckPa** [comment]
Does the WebTools installer no longer work?

I can update it if need be

**Comment by ChuckPa** [comment]
Given the status of WT,  I am not certain what will become of the effort.   
Discussion with @ukdtom makes it seem like a network retrieval issue.

@cswanghan 
I can create easy installers for any of @ukdtom 's packages you wish.   In a worst-case scenario,  do you have any preferences?

**Comment by jack061** [comment]
how can I handle this issue?  cannot extract wt.tar.gz to /volume2/Plex/Library/Application Support/Plex Media Server/Plug-ins/WebToos.bundle( error 0) @ukdtom

**Comment by ukdtom** [comment]
@ChuckPa ^^

**Comment by noproxy** [comment]
This is a network speed issue exactly.

The same issue occurs to me.
```
2019-10-19 18.04.30 CST: Updating from https://api.github.com/repos/ukdtom/WebTools.bundle/tarball/3.0.0
2019-10-19 18.05.00 CST: Error 0 while downloading WebTools
2019-10-19 18.05.00 CST: Download successful. Installation complete.
2019-10-19 18.05.00 CST: ERROR: Cannot extract wt.tar.gz to /volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/WebTools.bundle ( error 0 )
2019-10-19 18.05.00 CST: ERROR: Could not extract WebTools. Package installation aborted.
```

The downloading code is here: 
https://github.com/ukdtom/WTInstaller/blob/1e87c03480e2a263fcdea877c66e4dc51148d9f2/src/Synology/scripts/postinst#L100
https://github.com/ukdtom/WTInstaller/blob/1e87c03480e2a263fcdea877c66e4dc51148d9f2/src/Synology/scripts/postinst#L196

The ```-m $1``` parameters set a max download time ( 30 seconds ). But in bad network, it may cost more time.

I think the timeout limit should be larger. As for me, it cost `01:35` to download the whole archive.

**Comment by noproxy** [comment]
😉 I workaround by setting a proxy in Network Preferences. And then downloading was finished in 1 seconds.

Anyway, a fix is grateful and necessary for other users.

**Comment by ukdtom** [comment]
@ChuckPa  ^^

**Comment by chrisweetman** [comment]
I know this is old, but I had the same issue, paused downloads and then everything installed correctly, went to UAS and installed trackt addon, went into plex and configured via the settings/plug-ins and it appeared all was working...

go back into webtools and look at logs and can see errors, firstly
"ImportError: No module named apsw"
also
"File "/volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/exception_wrappers/libraries/apsw.py", line 16, in <module>"
and
"File "/volume1/Plex/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/exception_wrappers/libraries/playhouse/apsw_ext.py", line 24, in <module>"

I have the trakt plugin working on my new windows based plex server, but cannot get it working on sinology no matter what I try (armv7 cpu / DS2015XS).  I would really like to have my history uploaded from the synology as I have been running it for the last 5 years and have amassed a decent library and history through plex, would be a shame to lose all that data (I used to use kodi and centralised the database, that worked perfectly, I swapped as the new TVs natively support plex, but wouldn't support Kodi)

If anyone could shed some light on this or how to correct, I have been going round in circles online trying to fix, but I am a bit lost!

Cheers

Chris.
